compiler/codegen/OMRMachine.hpp

```diff
@@ -46,21 +47,79 @@ class OMR_EXTENSIBLE Machine
    {
    TR::CodeGenerator *_cg;
 
-public:
+   int16_t numLockedGPRs;
+   int16_t numLockedHPRs;
+   int16_t numLockedFPRs;
+   int16_t numLockedVRFs;
+
+   public:
 
    TR_ALLOC(TR_Memory::Machine)
 
-   Machine(TR::CodeGenerator *cg) :
-         _cg(cg)
+   Machine(TR::CodeGenerator *cg)
+      :
+      _cg(cg),
+      numLockedGPRs(-1),
+      numLockedHPRs(-1),
+      numLockedFPRs(-1),
+      numLockedVRFs(-1)
```

Comment:
I think it is because of the asserts we have on Power in OMRCodeGenerator.cpp which in a debug build will count the number of locked registers and validate that whatever is returned by `getNumberOfLockedRegisters` is actually correct. The counts there start at 0 hence you will never get false positives, i.e. you never initialized the numbers via `getNumberOfLockedRegisters` but it just so happens that you had no locked registers. This is still a bug.